bmc-log-collector/bmc-http-client.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"crypto/tls"
+	//"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"time"
+)
+
+// タイムアウトが必要？
+// Get from Redfish on iDRAC webserver
+func bmcClient(url string) ([]byte, error) {
+	fmt.Println("client start to ", url)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+
+	// タイムアウトの処理エラーのセットが欲しい
+	client := &http.Client{Timeout: time.Duration(10) * time.Second, Transport: tr}
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		slog.Error(fmt.Sprintf("%s", err))
+		return nil, err
+	}
+	req.SetBasicAuth(os.Getenv("BMC_USER"), os.Getenv("BMC_PASS"))
```

Comment:
Don't read environment variables here.
Read the variables in the main function and pass the user name and password as arguments.
Additionally, check for errors when environment variables are not set.